pkg/agent/metrics/metrics.go

```diff
@@ -20,13 +20,23 @@ import (
 	"time"
 
 	"github.com/prometheus/client_golang/prometheus"
+	"google.golang.org/grpc/status"
+	"sigs.k8s.io/apiserver-network-proxy/konnectivity-client/proto/client"
 )
 
 type Direction string
 
 const (
-	namespace = "konnectivity_network_proxy"
-	subsystem = "agent"
+	Namespace = "konnectivity_network_proxy"
+	Subsystem = "agent"
+
+	DialLatencyMetric         = "dial_duration_seconds"
+	ServerFailuresMetric      = "server_connection_failure_count"
```

Comment:
counters should be suffixed with `_total`.

Reply:
Not inlining the metric names makes it difficult to spot conventional naming errors like this one.

Reply:
Fixed.

Does k/k have any automatic metric name validations, that should be added to this repo? I half-remember this from  https://github.com/kubernetes/kubernetes/pull/113520

Reply:
We don't have it yet, but this is a good idea.